sql/type.go

```diff
@@ -314,7 +331,8 @@ func IsNull(ex Expression) bool {
 
 // IsNumber checks if t is a number type
 func IsNumber(t Type) bool {
-	return IsInteger(t) || IsDecimal(t)
+	_, ok := t.(numberTypeImpl)
```

Comment:
Might leave this alone for now -- these things can't be used in arithmetic and so on yet. This would cause some weird behavior if you did this before implementing this.

Speaking of which, you should have at least one test of using these new types in arithmetic expressions. It should do something reasonable (i.e., not panic, not give nonsensical results).